src/openrct2/world/Map.cpp

```diff
@@ -203,9 +203,9 @@ void map_set_tile_elements(int32_t x, int32_t y, TileElement* elements)
     gTileElementTilePointers[x + y * MAXIMUM_MAP_SIZE_TECHNICAL] = elements;
 }
 
-SurfaceElement* map_get_surface_element_at(int32_t x, int32_t y)
+SurfaceElement* map_get_surface_element_at(const TileCoordsXY& tileCoords)
```

Comment:
Remove the overload for `TileCoordsXY` it should only be used with `CoordsXY`.

Reply:
What about all of the places that use it, we'll convert to `CoordsXY`, doing the multiplications, or are they just wrong?
(Including the one I commented when opening the PR)

Reply:
As it stands its a bit confusing having the two versions when types can be easily converted between the two. `CoordsXY` should be preferred in almost all situations. I'm not really sure what is best because this function is probably one of the ones that is called a ridiculous amount of times per tick so it should probably made to be as simple as possible.